### DISCUSSION TITLE
Deep-linking more than one level results in unexpected child state

### DISCUSSION THREAD

**Discussion by ghost** [pointfreeco/swift-composable-architecture#1765] [Q&A]
In an app with navigation structured as follows...

```
          -> Intermediate A -> Leaf A
AppScene 
          -> Intermediate B -> Leaf B
```

...navigating to Leaf A, then handling a deep link intended to navigate to Leaf B, results in logs that indicate something "received a child action when child state was set to a different case"

The code that handles the deep link is as follows:

```
                case .openUrl(_):
                    // Starting from Intermediate A -> Leaf Scene A, hit the deep link tca20221220://nothing,
                    // pretend we parsed the URL here and decided we want to and drill down to IntermediateSceneB -> LeafSceneB
                    // It will complain that something "received a child action when child state was set to a different case"
                    state.route = .routeIntermediateB(IntermediateScene.State(
                        route: .routeLeaf(LeafScene.State(letterId: "B")),
                        letterId: "B"))
                    // This simpler case works:
                    //state.route = .routeIntermediateA(IntermediateScene.State(letterId: "B"))
                    return .none
```

There is a full app demonstrating this at [link to TCAQuestion0001 removed, outdated] (make sure you're on the TCAQuestion0001 branch for this, which should be the default).  To see the issue in the app, click the "Show Intermediate A" button, then the "Show Leaf" button, which will take you to Leaf Scene A.  Then background the app, and (one way or another, perhaps by making an appointment in the simulator's calendar with the URL) follow a deep link to tca20221220://nothing .

How can this `child state was set to a different case` error be prevented?  Is it possible to set a two-level route like this, directly on receiving a deep link?  The commented out "simpler case" of setting a one-level route works without complaint.

**Comment by ghost** [comment]
Is this possibly caused by the infamous "can't navigate deeper than 2 levels" bug in SwiftUI's `NavigationLink` API?

**Comment by mbrandonw** [comment]
Hi @wilsonmhpn, this is unfortunately just a bug with SwiftUI. The same would happen in vanilla SwiftUI without TCA. If you are linked deep enough into an app (such as drill-down > sheet), then trying to all-at-once reset it to a different navigation configuration just fails.

The fix is to help out SwiftUI by creating multiple steps to restore the navigation. You could first clear out navigation to bring it back to the root, pause a brief moment, and _then_ you can add the new drill-down+sheet.
